### PR TITLE
Remove `check`, `--explain`, `--clean`, `--generate-shell-completion` aliases

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR changes  `ruff <path>`, `ruff --explain`, `ruff --clean` and `ruff --generate-shell-completion` to error but continues to print an error message. The commands were deprecated in https://github.com/astral-sh/ruff/pull/10169 (v0.3.0)

Part of https://github.com/astral-sh/ruff/issues/10171

## Test Plan

I verified that:

* The JetBrains extension uses `ruff check`
* The LSP uses `ruff check`
* [python-lsp-ruff](https://github.com/python-lsp/python-lsp-ruff/blob/34acd6ed6daaaa3f2756a1a1cf1bfc2d76df7d0f/pylsp_ruff/plugin.py#L67) I think it uses `ruff check`

### CODE REVIEW

**Comment by dhruvmanila** [comment]
I'm looking at the Neovim ecosystem, already completed in `conform.nvim`:  https://github.com/stevearc/conform.nvim/pull/348

File: docs/configuration.md [hunk 1]
```diff
@@ -521,15 +521,21 @@ Ruff: An extremely fast Python linter.
...
-  help     Print this message or the help of the given subcommand(s)
+  check                      Run Ruff on the given files or directories
+                                 (default)
+  rule                       Explain a rule (or all rules)
+  config                     List or describe the available configuration
+                                 options
+  linter                     List all supported upstream linters
+  clean                      Clear any caches in the current directory and
+                                 any subdirectories
+  generate-shell-completion  Generate shell completion
```
Comment (zanieb) [inline docs/configuration.md:532]: I feel like we've kept this hidden previously? Definitely nicer indentation when hidden.

Comment (MichaReiser) [reply docs/configuration.md:532]: Nice catch. Yeah, I deleted a little too much. Thank you

**Comment by MichaReiser** [comment]
I changed the implementation to error instead of removing the aliases entirely.

**Pull request merged**